### ISSUE TITLE
aFBA RetroArch is not working

### ISSUE THREAD

**Issue by ReCoJav** [ResetCollection/support#40]
A user has contacted me about aFBA not working. Need to investigate, and if there is an issue, fix it!

**Comment by Kinghumpy** [comment]
Any luck on getting fba or fbn working. I get a black screen and app stop responding error on launch of a game.

**Comment by ReCoJav** [comment]
Howdy!

So sorry about this! Really sucks that the final burns aren't working
right. I'm going to spend some time to thoroughly test them and fix
whatever issue is going on. I do really apologize for this one piece of bad
news, but it'll probably be a while before I fix this (like a few
weeks)... I have some challenges going on here due to the times. Trying to
move family around the states and their dogs during these times! No fun.

I do promise it'll be fixed! It is on my todo list for sure.

One question for you, which version of RetroArch are you using?

Okay, thanks! And take care!
-Jav

**Comment by Kinghumpy** [comment]
Hey brother dont stress it. You have a good thing going here. You will get the kinks worked out in time. I'm a long time launchbox supporter and with the flop of there android app they recommended your app and man you take it to the next level. Good luck to ya. My retroarch version is 1.8.5.

**Comment by seeknom** [comment]
I've been doing a lot reading about FBA today, it seems like it's very prone to compatibility issues due to rom archive format/structure. There's a couple of NeoGeo games I've been trying to get working (Sonic Wings series) and I was getting the same lockups/black screen Kinghumpy reported. I finally found some ROM versions that work with Retroarch (latest 32-bit), but unfortunately the 2 cores that support them are not supported by ReCo. If possible, maybe these RA cores could be added?

Arcade (FB Alpha 2012)     (as opposed to "FB Alpha 2012 Neo Geo" that doesn't work for me).
Arcade (FinalBurn Neo) 

Great work Jav!

**Comment by javierarboleda** [comment]
Thanks for the note, @seeknom 

I think I'll be spending some time investigating these cores very very soon 👍

**Comment by ReCoJav** [comment]
Thanks for the note, @seeknom 

I think I'll be spending some time investigating these cores very very soon 👍

**Comment by ReCoJav** [comment]
@seeknom 

Those cores should be in the Arcade system. 
Arcade (FB Alpha 2012)
Arcade (FinalBurn Neo)

Mind checking again?

I'll add a zip of a video showing setting a Neo Geo game in RESET with Final Burn Neo and launching it.

[final_burn_neo_video.zip](https://github.com/ResetCollection/support/files/4648883/final_burn_neo_video.zip)

**Comment by ReCoJav** [comment]
@Kinghumpy forgot to tag you also in comment above, sorry!

**Comment by seeknom** [comment]
Yep, I see the cores. Not sure how or why I thought they were missing, but I have no evidence it's a bug in ReCo and not a mistake on my part. Thanks, and apologies for the erroneous report!

**Comment by Kinghumpy** [comment]
So I watched your video. I have tried it and I have the same issue. I have both the 32 and 64 bit retroarch versions that are up to date and I have all the cores for both versions. Tried all of the cores with both 32 and 64 bit and I get black screens on both. Sucks because I'm using arc browser separate to play my arcade games and reco for all my others.

**Comment by Kinghumpy** [comment]
So I don't know if this has anything to do with it but I deleted all of my retroarch apps and cores and set the emulator to let your app direct me to Google to download it. I downloaded the 1 core only. Then I went to set the emulator and picked the fbneo_libretro_android.so and selected 32 bit first option and when I launch a game it tells me I don't have fbalpha_libretro_android.so installed. That's the wrong core in the error? Hope this helps

**Comment by ReCoJav** [comment]
> So I don't know if this has anything to do with it but I deleted all of my retroarch apps and cores and set the emulator to let your app direct me to Google to download it. I downloaded the 1 core only. Then I went to set the emulator and picked the fbneo_libretro_android.so and selected 32 bit first option and when I launch a game it tells me I don't have fbalpha_libretro_android.so installed. That's the wrong core in the error? Hope this helps

@Kinghumpy 

Thanks for doing this extra work to try to get RetroArch to work with RESET. Sorry it's still not working out!

Just to make sure I'm clear, after launching the game, is it RetroArch that is giving you the message that you don't have "fbalpha_libretro_android.so" installed?

This is just so weird. I'm looking at my code, and there is no way RESET would be sending another core in the data for RetroArch to use than "fbneo_libretro_android.so" .... But having said that, weird program stuff can always happen. But this shouldn't be happening.

Yeah, please let me know if it's RetroArch giving the error message or RESET.

Thanks.

**Comment by Kinghumpy** [comment]
Here is a video

**Comment by Kinghumpy** [comment]
Here is a video

**Comment by Kinghumpy** [comment]
Here is a video

[Recording_2020-05-25-10-16-20.zip](https://github.com/ResetCollection/support/files/4677651/Recording_2020-05-25-10-16-20.zip)

**Comment by Kinghumpy** [comment]
Ok. I found the issue. So if I go to a single game and set the emulator to fbneo that game will work perfect. So my problem is when I select emulator for all games for arcade and pick fbneo it will default to fbalpha instead.

**Comment by ReCoJav** [comment]
> Ok. I found the issue. So if I go to a single game and set the emulator to fbneo that game will work perfect. So my problem is when I select emulator for all games for arcade and pick fbneo it will default to fbalpha instead.
@Kinghumpy 

OH WTF, that is a very edge and hard to find bug! Thank you so much for finding that!

I'll take a look into why this is happening.

Thanks again, and sorry it took time out of your life to figure out this weird bug! If you ever find me out and about, I owe you a beer/soda/hamburger/vegan burger (can't assume, lol).

**Comment by Kinghumpy** [comment]
No problem. Hopefully this will help point you in the right direction. With everyone's help you will have the #1 front end for android. Honestly it's already better than arc browser imo. Emulation is life and your app completes me
